metadata_layer.py

```diff
@@ -590,7 +590,7 @@ def set_default_rights_uri(self, data_source):
         for format in self.formats:
             if format.link:
                 link = format.link
-                if self.rights_uri == None and link.rel == Hyperlink.OPEN_ACCESS_DOWNLOAD:
+                if self.rights_uri in (None, RightsStatus.UNKNOWN) and link.rel == Hyperlink.OPEN_ACCESS_DOWNLOAD:
```

Comment:
The other possibility I can think of is that the source OPDS feed is mistaken or lying about the open-access status of the work. That is, the feed gives a link with rel="open-access" that points to a copy of a book that's actually not distributable. But that's a possibility even when the license is explicitly stated, so I don't think we can address that problem here.